src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -201,8 +453,259 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
 
+    /**
+     * Returns the name of the query parser to use when parsing
+     * {@link SimpleQueryStringBuilder} objects from JSON.
+     */
     @Override
     public String queryId() {
         return NAME;
     }
+
+    /**
+     * Reads a {@link SimpleQueryStringBuilder} object from the given input
+     * stream.
+     * 
+     * @param in Stream to read the object from.
+     * @return the object read from the stream.
+     * */
+    @Override
+    public SimpleQueryStringBuilder readFrom(StreamInput in) throws IOException {
+        String text = in.readOptionalString();
+        float boost = in.readFloat();
+        int size = in.readInt();
+        Map<String, Float> fields = new HashMap<>();
+        for (int i = 0; i < size; i++) {
+            String field = in.readString();
+            Float weight = in.readFloat();
+            fields.put(field, weight);
+        }
+
+        int flags = in.readInt();
+        String analyzer = in.readOptionalString();
+
+        Operator operator = null;
+        Integer operatorId = in.readOptionalVInt();
+        if (operatorId != null) {
+            switch (operatorId) {
+            case 1:
+                operator = Operator.OR;
+                break;
+            case 2:
+                operator = Operator.AND;
+                break;
+            }
+        }
+
+        Boolean lowercaseExpandedTerms = in.readOptionalBoolean();
+        Boolean lenient = in.readOptionalBoolean();
+        Boolean analyzeWildcards = in.readOptionalBoolean();
+
+        Locale locale = null;
+        if (in.readBoolean()) {
+            String localeStr = in.readString();
+            locale = Locale.forLanguageTag(localeStr);
+        }
+        String queryName = (in.readOptionalString());
+        String minimumShouldMatch = in.readOptionalString();
+        return new SimpleQueryStringBuilder(text, boost, fields, analyzer, queryName, operator, minimumShouldMatch, flags, locale,
+                lowercaseExpandedTerms, lenient, analyzeWildcards);
+    }
+
+    /**
+     * Writes the current {@link SimpleQueryStringBuilder} object to the given
+     * output stream.
+     * 
+     * @param out The stream to write the object to.
+     **/
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalString(queryText);
+        out.writeFloat(boost);
+        out.writeInt(fieldsAndWeights.size());
+        for (Map.Entry<String, Float> entry : fieldsAndWeights.entrySet()) {
+            out.writeString(entry.getKey());
+            out.writeFloat(entry.getValue());
+        }
+        out.writeInt(flags);
+        out.writeOptionalString(analyzer);
+        out.writeBoolean(defaultOperator != null);
+        if (defaultOperator != null) {
+            switch (defaultOperator) {
+            case OR:
+                out.writeVInt(1);
+                break;
+            case AND:
+                out.writeVInt(2);
+                break;
+            }
```

Comment:
we can simply make the id part of the Operator enum, and add a parseFromInt method to it too that gets called from readFrom. writeTo just does out.writeVInt(operator.id())